lib/datadog/appsec/actions_handler.rb

```diff
@@ -19,7 +22,33 @@ def interrupt_execution(action_params)
         throw(Datadog::AppSec::Ext::INTERRUPT, action_params)
       end
 
-      def generate_stack(_action_params); end
+      def generate_stack(action_params)
+        return unless Datadog.configuration.appsec.stack_trace.enabled
+
+        context = AppSec.active_context
+        if context.nil? || context.trace.nil? && context.span.nil?
+          Datadog.logger.debug { 'Cannot find trace or service entry span to add stack trace' }
+          return
+        end
+
+        config = Datadog.configuration.appsec.stack_trace
+
+        # Check that the sum of stack_trace count in trace and entry_span does not exceed configuration
+        span_stack = ActionsHandler::StackTraceInMetastruct.create(context.span&.metastruct)
+        trace_stack = ActionsHandler::StackTraceInMetastruct.create(context.trace&.metastruct)
+        return if config.max_collect != 0 && span_stack.count + trace_stack.count >= config.max_collect
```

Comment:
do we want to avoid stack trace collection if the stack is longer than `max_collect`, or do we want to truncate the stack?